### DISCUSSION TITLE
OpenRecall Roadmap

### DISCUSSION THREAD

**Discussion by tvturnhout** [openrecall/openrecall#9] [Polls]
Vote on your favorite feature to be added next

**Comment by 1472583610** [comment]
One feature I could definitely use would be a list of files (documents) opened in that particular program at that particular time. In the ideal case, each item in the list would be clickable to open it straight away in the associated app.


BTW, if it's of any interest, here is my reasoning for each feature in the poll:

**1. Give time scroll bar colors to distinguish between apps:** A QoL improvement, so useful, but not a priority.

**2. Add text to screenshot results for selecting and copy/pasting:** I voted for this one, I would use this feature primarily to search for text contained in arbitrary binary files.

**3. Add audio recording and search in audio:** Also very useful, though not as high priority for me as nr. 2 or easily accessible list of files (my suggestion of a feature above).

**4. Add text to image for automated image description and improve visual recall:** I'm not sure I'd find a proper use case for this one.

**5. Dark theme:** Always good to have, I always use it, but it's not a priority.

**6. Exclude apps for privacy:** Also nice to have, but as long as the entire project is privacy-safe, it doesn't matter it watches all activity. Could be an optimization feature, though. No need to invest computing power and energy (especially on battery-powered devices) on apps that aren't needed by the user.

**7. Possibility to locally encrypt history database:** Performance concerns here and low priority for me, I'm confident in my ability to keep my data safe, even unencrypted.

**Reply by tvturnhout** [reply]
Thanks for you reply, appreciate the feedback. We have actively taken your feedback to re-order some of our to-do's 👍

**Comment by TanMan1217** [comment]
I was coming here to strongly suggest encrypting your database, but I see it's already on the feature list, so I voted for it. However, even if you encrypt the database, you still need to protect the web interface with a user and password. Without both the encryption and the authentication, if someone gets access to your device, they'll be able to easily see EVERYTHING that you ever did. That's a BIG problem.

**Reply by koenvaneijk** [reply]
Yes, fully agree with you. There are multiple vectors obviously: someone getting physical access to your machine when it is unattended and unlocked, malware that perhaps achieves certain elevations, ...  The data should ideally be encrypted in the history database and decrypted ad-hoc in the web interface (if encryption key is provided). We're investigating to what extent we could use hardware keys, or TPM or Secure Enclave, ... for this.

**Comment by formica45** [comment]
It would be nice to search what I heard a day ago. Thumbs up for audio search - thanks

**Comment by auphof** [comment]
Really like what I am seeing here with good planning around roadmap and feature docs, Please consider adding [MulitMonitor support](https://github.com/openrecall/openrecall/issues/15) and encrypted data https://github.com/openrecall/openrecall/issues/12  to this voting list , MultiMontor would be my top priority with a very close second being encrypted data . I have 3 monitors, without this I would not catpture 2/3rds of my timeline, Thank you for your superb efforts

**Reply by koenvaneijk** [reply]
We have added multi-monitor support in v0.3!

**Comment by DmacMcgreg** [comment]
Just want to agree here that this is one of the more advanced recall/rewind clones out there so far. I just want to add that the code is clean, simple and modular which should really help in adding new features quickly. Happy to jump in here and help out where possible.

**Comment by GerardPolloRebozado** [comment]
Also about hiding certain apps for privacy it would be nice to hide other things like incognito mode when being on browser or when having inputs of sensible data like password or credit cards

**Comment by m5x5** [comment]
It'd be great if we had a delete button for screenshots just in case sensitive data is captured ;)

**Comment by Elaman0117** [comment]
Fetures are great but I am a full-time dark mode user, so... I'd like to protect my eyes first.🥴

**Comment by SyndicatedPillbug** [comment]
I'd really like to see native Installers for Linux, Mac, and Windows - and while I see two of those on the roadmap, I'm a bit worried that us Linux users who really desire a streamlined install process will get left in the dust. Sure, some Linuxites love diving into Terminal for everything... but Flatpak is going to be easier for most people to understand.

**Comment by aholeinthehead** [comment]
Historical screenshots can end up being huge.  I'd love to see a feature that archives older screenshots.  Keep the index data and thumbnails, but attach to an S3 back end to store encrypted archives.  Sometimes I find out months after I broke something, and it's helpful to scroll back in time to find out exactly what I did.

**Comment by erkinalp** [comment]
Reverse RDP capture mode (some people think Windows Recall works like that, so I made a design to make it real): #117

**Reply by erkinalp** [reply]
### Real Windows Recall is not a reverse RDP capture.
